docs/DeveloperGuide.md

```diff
@@ -1,22 +1,145 @@
-# Developer Guide
+# JavaNinja Developer Guide
+Java Ninja is a command-line interface (CLI) learning tool focused on helping beginner programmers enhance their understanding of fundamental programming concepts.
+
+- [Design & implementation](#design-&-implementation)
+- [Product scope](#product-scope)
+    - [Target user profile](#target-user-profile)
+    - [Value proposition](#value-proposition)
+- [User Stories](#user-stories)
+- [Non-Functional Requirements](#non-Functional-Requirements)
+- [Glossary](#Glossary)
+- [Instructions for manual testing](#Instructions-for-manual-testing)
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+- [Apache Commons Lang Library](https://commons.apache.org/proper/commons-lang/) - Used for string utilities.
+- [JUnit 5](https://junit.org/junit5/) - Used for testing.
+- Code adapted from [AddressBook-Level3](https://github.com/se-edu/addressbook-level3).
+
+JavaNinja uses the following tools for development:
+- JUnit - Used for testing.
+- Gradle - Used for build automation.
 
 ## Design & implementation
+  
+### Design 
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+#### Architecture 
+
+Main components of the architecture:
+
+`Main` (consisting of class `JavaNinja`) is in charge of the app launch and shut down. 
+1. At app launch, it initialises the components in the correct order and connects them with each other
+2. At shut down, it shuts down the other components and invokes cleanup methods 
+
+The bulk of the app's work is done by the respective components: 
+1. `Cli`: It handles the user interface of the app 
+2. `Parser`: The command executor 
+3. `QuizManager`: Holds the data of the App in memory
+4. `Storage`: Reads data from, and writes data to, the hard disk.
+
+`Commons` represent a collection of classes used by other components.
+
+How the architecture components interact with each other: 
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user issues the command `add Flashcard /q what's 2+3 /a 5`
+
```

Comment:
Since you gave a sequence diagram for add Flashguard, you should have a flow of events (for that feature) above it to explain what goes on.